lxd/patches.go

```diff
@@ -1346,6 +1347,44 @@ func patchRemoveCoreTrustPassword(_ string, d *Daemon) error {
 	return nil
 }
 
+func patchEntityTypeInstanceSnapshotTypoFix(name string, d *Daemon) error {
+	for {
```

Comment:
I've had a good think about the scenarios we need to account for here and I think we should actually run this patch on all cluster members.

Each member will only delete the trigger if it exists, and will have added the new trigger when loading the database, so I don't think we're in danger of missing anything. The worry would be that a member that has not yet been updated reloads the LXD daemon and reapplies the old trigger. However, we're less in danger of this becoming an issue if we apply the patch on all members because once they're all updated we can be certain that the old trigger is deleted. In contrast, if we only run this patch on the leader then we're in danger of another cluster member that has not yet been updated reloading the LXD daemon and re-adding the old trigger.

The upside is the patch will be much simpler. It should look like https://github.com/canonical/lxd/blob/f9b1a03ef3c5efdb2ec32c624e26f24eecaf296f/lxd/patches.go#L1334-L1347 but removing the old trigger.

Does this make sense? Could you sanity check this please @tomponline?

Reply:
That makes sense to me, I'll apply the changes after the go ahead from @tomponline

Reply:
Because triggers are setup on all members, the patch that affects triggers also has to be setup on all members right?

Reply:
Yes, exactly. Since triggers are associated with each database member, applying the patch to all members should ensure consistency.